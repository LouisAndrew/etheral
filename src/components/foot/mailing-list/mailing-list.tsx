
import React from 'react'
// import styling libs
// import local components

type Props = {

}

const MailingList: React.FC<Props> = () => {

	return <></>
}

export {MailingList}
