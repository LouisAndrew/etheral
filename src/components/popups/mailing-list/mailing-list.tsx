import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';
import Img from 'gatsby-image';

import { Flex, Box, Text, Heading } from 'rebass';
import { Input } from '@rebass/forms';
import { Icon } from '@iconify/react';
import closeFill from '@iconify/icons-ri/close-fill';

type Props = {
    closeModal: () => void;
};

const MailingList: React.FC<Props> = ({ closeModal }) => {
    const data = useStaticQuery(graphql`
        query {
            imgS: file(relativePath: { eq: "mail-popup.jpg" }) {
                childImageSharp {
                    fixed(height: 300, width: 300, quality: 100) {
                        ...GatsbyImageSharpFixed
                    }
                }
            }
            imgL: file(relativePath: { eq: "mail-popup.jpg" }) {
                childImageSharp {
                    fixed(height: 400, width: 400, quality: 100) {
                        ...GatsbyImageSharpFixed
                    }
                }
            }
        }
    `);

    const { imgS, imgL } = data as any;
    if (imgS && imgL) {
        const sources = [
            {
                ...imgL.childImageSharp.fixed,
                media: '(min-height: 400px) and (min-width: 900px)',
            },
            imgS.childImageSharp.fixed,
        ];

        return (
            <Flex
                flexDirection={['column', 'row']}
                sx={{ position: 'relative' }}
            >
                <Box
                    onClick={closeModal}
                    role="button"
                    aria-label="close-modal-button"
                    sx={{
                        position: 'absolute',
                        top: [1],
                        right: [1],
                        zIndex: 2000,
                        '& > svg': {
                            height: [24],
                            width: [24],
                            path: {
                                fill: ['#fff', '#000'],
                            },
                        },
                    }}
                >
                    <Icon icon={closeFill} id="close-popup" />
                </Box>
                <Img fixed={sources} alt="bracelet" />
                <Flex
                    bg="white.0"
                    p={[4, 6, 6, 8]}
                    flexDirection="column"
                    justifyContent="space-evenly"
                    css={`
                        height: fit-content;
                        width: 300px;

                        @media screen and (min-width: 27em) {
                            height: 300px;
                        }

                        @media screen and (min-width: 900px) {
                            height: 400px;
                            width: 400px;
                        }
                    `}
                >
                    <Box>
                        <Heading
                            as="h3"
                            variant="h3"
                            id="subscribe-newsletter-popup"
                        >
                            SUBSCRIBE TO OUR NEWSLETTER
                        </Heading>
                        <Text variant="body" mt={[3]}>
                            To get your 10% off your first purchase
                        </Text>
                    </Box>
                    <Box as="form" my={[6]}>
                        <Input
                            type="email"
                            aria-labelledby="subscribe-newsletter-popup"
                            placeholder="Enter your email address here"
                            variant="variants.textInput"
                            width={['100%']}
                            maxWidth="100%"
                        />
                        <Input
                            type="submit"
                            variant="buttons.primary"
                            value="GET 10% OFF"
                            sx={{ border: 'none', mt: [5] }}
                        />
                    </Box>
                </Flex>
            </Flex>
        );
    }

    return null;
};

export { MailingList };
