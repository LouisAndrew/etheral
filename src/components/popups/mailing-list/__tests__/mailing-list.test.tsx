import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import userEvent from '@testing-library/user-event';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { MailingList } from '../mailing-list';

import * as Gatsby from 'gatsby';
const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');

describe('MailingList', () => {
    const mockCloseModal = jest.fn(() => {});

    const Element = <MailingList closeModal={mockCloseModal} />;

    beforeAll(() => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(), // deprecated
                removeListener: jest.fn(), // deprecated
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            })),
        });
    });

    afterEach(cleanup);
    beforeEach(() => {
        useStaticQuery.mockImplementationOnce(() => ({
            imgXS: {
                childImageSharp: {
                    fixed: {},
                },
            },
            imgS: {
                childImageSharp: {
                    fixed: {},
                },
            },
            imgL: {
                childImageSharp: {
                    fixed: {},
                },
            },
        }));
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    it('should close modal when user clicked the close button correctly', () => {
        const { queryByLabelText } = render(Element);

        const CloseButton = queryByLabelText('close-modal-button');

        if (CloseButton) {
            expect(CloseButton).toBeInTheDocument();
            userEvent.click(CloseButton);

            expect(mockCloseModal).toHaveBeenCalled();
        } else {
            fail();
        }
    });

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
