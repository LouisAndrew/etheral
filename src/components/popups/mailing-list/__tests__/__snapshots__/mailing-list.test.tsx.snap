// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailingList matches snapshot 1`] = `
<div
  className="css-1tw1h9e"
>
  <div
    aria-label="close-modal-button"
    className="css-9tbm8a"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [
            Class {
              "_dispatchInstances": null,
              "_dispatchListeners": null,
              "_targetInst": null,
              "altKey": null,
              "bubbles": null,
              "button": null,
              "buttons": null,
              "cancelable": null,
              "clientX": null,
              "clientY": null,
              "ctrlKey": null,
              "currentTarget": [Function],
              "defaultPrevented": null,
              "detail": null,
              "dispatchConfig": null,
              "eventPhase": null,
              "getModifierState": [Function],
              "isDefaultPrevented": [Function],
              "isPropagationStopped": [Function],
              "isTrusted": null,
              "metaKey": null,
              "movementX": [Function],
              "movementY": [Function],
              "nativeEvent": null,
              "pageX": null,
              "pageY": null,
              "relatedTarget": [Function],
              "screenX": null,
              "screenY": null,
              "shiftKey": null,
              "target": null,
              "timeStamp": [Function],
              "type": null,
              "view": null,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    role="button"
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M12 10.586l4.95-4.95l1.414 1.414l-4.95 4.95l4.95 4.95l-1.414 1.414l-4.95-4.95l-4.95 4.95l-1.414-1.414l4.95-4.95l-4.95-4.95L7.05 5.636z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      id="close-popup"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <div
    className=" gatsby-image-wrapper"
    style={
      Object {
        "display": "inline-block",
        "height": undefined,
        "overflow": "hidden",
        "position": "relative",
        "width": undefined,
      }
    }
  >
    <picture>
      <source
        media="(min-height: 400px) and (min-width: 900px)"
      />
      <source />
      <img
        alt="bracelet"
        loading="lazy"
        onLoad={[Function]}
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 0,
            "position": "absolute",
            "top": 0,
            "transition": "opacity 500ms",
            "width": "100%",
          }
        }
      />
    </picture>
    <noscript
      dangerouslySetInnerHTML={
        Object {
          "__html": "<picture><source media=\\"(min-height: 400px) and (min-width: 900px)\\" srcset=\\"undefined\\" /><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"\\" alt=\\"bracelet\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
        }
      }
    />
  </div>
  <div
    className="css-146twly"
  >
    <div
      className="css-15fz7r2"
    >
      <h3
        className="css-qjyand"
        id="subscribe-newsletter-popup"
      >
        SUBSCRIBE TO OUR NEWSLETTER
      </h3>
      <div
        className="css-jmlio4"
      >
        To get your 10% off your first purchase
      </div>
    </div>
    <form
      className="css-rh7pv8"
    >
      <input
        aria-labelledby="subscribe-newsletter-popup"
        className="css-1pr8502"
        placeholder="Enter your email address here"
        type="email"
      />
      <input
        className="css-1spzla3"
        type="submit"
        value="GET 10% OFF"
      />
    </form>
  </div>
</div>
`;
